vortex-array/src/arrays/chunked/compute/cast.rs

```diff
@@ -65,4 +68,31 @@ mod test {
             &[0u64, 1, 2, 3],
         );
     }
+
+    #[rstest]
+    #[case(ChunkedArray::try_new(
+        vec![buffer![0u32, 1, 2].into_array(), buffer![3u32, 4].into_array()],
+        DType::Primitive(PType::U32, Nullability::NonNullable)
+    ).unwrap().into_array())]
+    #[case(ChunkedArray::try_new(
+        vec![
+            buffer![-10i32, -5, 0].into_array(),
+            buffer![5i32, 10].into_array()
+        ],
+        DType::Primitive(PType::I32, Nullability::NonNullable)
+    ).unwrap().into_array())]
+    #[case(ChunkedArray::try_new(
+        vec![
+            PrimitiveArray::from_option_iter([Some(1.5f32), None, Some(2.5)]).into_array(),
+            PrimitiveArray::from_option_iter([Some(3.5f32), Some(4.5)]).into_array()
+        ],
+        DType::Primitive(PType::F32, Nullability::Nullable)
+    ).unwrap().into_array())]
+    #[case(ChunkedArray::try_new(
+        vec![buffer![42u8].into_array()],
+        DType::Primitive(PType::U8, Nullability::NonNullable)
+    ).unwrap().into_array())]
+    fn test_cast_chunked_conformance(#[case] array: crate::ArrayRef) {
```

Comment:
Meta point: You should be able to express all of these with a common set of arrays that you want to convert to the target encoding and then run the test on and compare to the result of applying it on the original